mobile/library/cc/engine_builder.cc

```diff
@@ -152,6 +159,13 @@ void XdsBuilder::build(envoy::config::bootstrap::v3::Bootstrap* bootstrap) const
     list->add_patterns()->mutable_safe_regex()->set_regex("sds\\..*");
     list->add_patterns()->mutable_safe_regex()->set_regex(".*\\.ssl_context_update_by_sds");
   }
+
+  Grpc::forceRegisterDefaultGoogleGrpcCredentialsFactory();
```

Comment:
question: i'm not sure if these are needed if we include the targets in the build file as you've done in https://github.com/envoyproxy/envoy/pull/30151/files#diff-a1b29499dbdcd7ba89a7051f1f7b1c7f347a86841f6bb34906cfc293f7f58486R52.  it may still be required, i'm just not 100% sure.  can we try removing L164-L168 and re-running to see if it was required or not?

Reply:
You're right. We don't need that. Simply having the `BUILD` dependencies is sufficient because the extensions have `alwayslink = True`. I have removed the force registration code.